src/components/fields/edit/LocationIdEditor.jsx

```diff
@@ -36,46 +36,51 @@ export default function LocationIdEditor(props) {
     return result;
   }
 
-  function collapseChoices(choices, depth) {
-    const result = choices.map(choice => {
-      const numDescendants = getNumDescendents(choice);
-      const numDescendantsAsString = numDescendants
-        ? ` (${numDescendants})`
-        : '';
-      if (!choice.locationID) {
-        return {
-          depth,
-          name: choice.name + numDescendantsAsString,
-          id: choice.id,
-        };
+  const collapsedChoices = useMemo(
+    () => {
+      function collapseChoices(choices, depth) {
+        const result = choices.map(choice => {
+          const numDescendants = getNumDescendents(choice);
+          const numDescendantsAsString = numDescendants
+            ? ` (${numDescendants})`
+            : '';
+          if (!choice.locationID) {
+            return {
+              depth,
+              name: choice.name + numDescendantsAsString,
+              id: choice.id,
+            };
+          }
+          const subchoices = [
+            {
+              depth,
+              name: choice.name + numDescendantsAsString,
+              id: choice.id,
+            },
+          ];
+          subchoices.push(
+            collapseChoices(choice.locationID, depth + 1),
+          );
+          return subchoices;
+        });
+        return result.flat(100);
       }
-      const subchoices = [
-        {
-          depth,
-          name: choice.name + numDescendantsAsString,
-          id: choice.id,
-        },
-      ];
-      subchoices.push(collapseChoices(choice.locationID, depth + 1));
-      return subchoices;
-    });
-    return result.flat(100);
-  }
+
+      return collapseChoices(get(schema, 'choices', []), 0);
+    },
+    [get(schema, 'choices.length')],
+  );
+
+  const currentRegionArray = value
+    ? collapsedChoices.filter(choice => get(choice, 'id') === value)
+    : null;
+  const currentRegion = get(currentRegionArray, [0], null);
 
   return (
     <FormCore schema={schema} width={width}>
       <Autocomplete
-        value={
-          value
-            ? get(
-                get(schema, 'choices').filter(
```

Comment:
the biggest issue was here: the collapsed choices needed to be filtered, wasn't working on the nested object